lib/metadata/acl.js

```diff
@@ -14,10 +14,21 @@ const acl = {
 
     addObjectACL(bucket, objectKey, objectMD, addACLParams, params, log, cb) {
         log.trace('updating object acl in metadata');
-        // eslint-disable-next-line no-param-reassign
-        objectMD.acl = addACLParams;
-        metadata.putObjectMD(bucket.getName(), objectKey, objectMD, params, log,
-            cb);
+        function _didNotChange(grant) {
+            return objectMD.acl[grant].length === addACLParams[grant].length
+                && objectMD.acl[grant].every(value => addACLParams[grant].includes(value));
+        }
+        const cannedChanged = objectMD.acl.Canned !== addACLParams.Canned;
+        if (cannedChanged || !_didNotChange('FULL_CONTROL') || !_didNotChange('WRITE_ACP')
+            || !_didNotChange('READ') || !_didNotChange('READ_ACP')) {
+            /* eslint-disable no-param-reassign */
+            objectMD.acl = addACLParams;
```

Comment:
this _replaces_ the ACL : so it can actually also remove grants.... and is not consistent with the check in line 19 (`newAcl.includes(oldAcl)`)